File: test/sort.py
def selectionSort(lst):
    for i in range(0,len(lst)-1):
        minIndex = i
        for j in range(i+1,len(lst)):
            if lst[j] < lst[minIndex]:
                minIndex = j
        if minIndex != i :
            lst[minIndex],lst[i] = lst[i],lst[minIndex]

def bubbleSort(lst):
    for i in range(0, len(lst)-1):
        swap = 0
        for j in range(len(lst) - 1 , i ,-1):
            if lst[j] < lst[j - 1]:
                swap = 1
                lst[j],lst[j - 1] = lst[j - 1],lst[j]
        if swap == 0:
            break
        

def insertionSort(lst):
    for i in range(1, len(lst)):
        temp = lst[i]
        j = i - 1
        while j >= 0 and lst[j] > temp:
            lst[j+1] = lst[j]
            j = j-1
        lst[j+1] = temp

def main():
    #['Vijaya', 'Sanvi', 'Ruby', 'Zafar', 'Maya', 'Anya']
    #['501','1999','250','504','509','501']
    #[501,1999,250,504,509,501]
    lst = eval(input(" Enter a list :  "))
    print('Sorted List') 
    bubbleSort(lst)
    print(lst)
if __name__ == '__main__' :
    main()

File: test/sort.cpp

//  Problem 1 SELECTION SORT 
//         We have to arrange an array in increasing order 
//         if user enters 
//          {51,34,26,3,24,45} --> {3,24,26,34,45,51}


/*

void selection_sort(int arr[],int n){

    for(int i = 0; i < n-1 ; i++){
        int min = i;
        for(int j = i; j < n ; j++){
            if (arr[j] < arr[min]){
                min = j;
            }
        }
            cout<<"\nMinimum element found at "<<min<<" position which is "<<arr[min]<<"\n";
            //  Now swapping the elements
            cout<<" Now swapping "<<arr[i]<<" with "<<arr[min]<<"\n";
            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
            cout<<"     Now array is : ";
            for(int i = 0 ; i < n ; i++) cout<<arr[i]<<" ";
    }

}
*/


# include<iostream>
using namespace std;

void selectionSort(int arr[],int n){
    for(int i = 0 ; i < n -1 ; i++){
        int minIndex = i;
        for(int j = i + 1 ; j < n ; j++){
            if(arr[j] < arr[minIndex]){
                minIndex = j;
            }
        }
        if(minIndex != i){
            swap(arr[i],arr[minIndex]);
        }
    }
}

// void bubbleSort(int arr[],int n){
//     for(int i = 0 ; i < n - 1 ; i++){
//         int srt = 0;
//         for(int j = n ; j > i ; j--){
//             if (arr[j] < arr[j-1]){
//                 srt = 1;
//                 swap(arr[j],arr[j-1]);
//             }
//         }
//         if(srt == 0){
//             cout<<"\n Given array is already sorted \n ";
//             break;
//         }
//     }
// }
void bubbleSort(int arr[],int n){
    for(int i = n-1 ; i >= 1 ; i--){
        for(int j = 0 ; j <= i-1 ; j++){
            if( arr[j] > arr[j+1]){
                swap(arr[j],arr[j+1]);
            }
        }
    }
}
void insertionSort(int arr[],int n){
    for(int i = 1 ; i < n ; i++){
        int temp = arr[i];
        int j = i - 1;
        while(j >= 0 && arr[j] > temp){
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = temp;
    }
}

int main(){
    int n;
    cout<<"Enter the size of an array you will give : " ;
    cin>>n ;
    int arr[n];
    for(int i = 0 ; i < n ; i++) cin>>arr[i];
    cout<<"\n Entered array is : ";

    for(int i = 0 ; i < n ; i++) cout<<arr[i]<<" ";

    // selectionSort(arr,n);
    // insertionSort(arr,n);
    bubbleSort(arr,n);

    cout<<"\n Sorted array is : ";

    for(int i = 0 ; i < n ; i++) cout<<arr[i]<<" ";
    return 0;


}
